packages/staking-contracts/src/safe-transaction-proposer.ts

```diff
@@ -0,0 +1,134 @@
+import SafeApiKit from "@safe-global/api-kit";
+import Safe from "@safe-global/protocol-kit";
+import { MetaTransactionData, OperationType } from "@safe-global/types-kit";
+import { ethers } from "ethers";
+import { Hex } from "viem";
+
+import { abi } from "./abi.js";
+import { Network, getChainForNetwork } from "./network.js";
+import { AllocationResult, RewardsAllocator } from "./rewards-allocator.js";
+
+/**
+ * Configuration for the Safe transaction proposer
+ * @param safeAddress The address of the Safe
+ * @param rpcUrl The URL of the RPC provider
+ * @param apiKey The API key for the Safe
+ * @param chainId The chain ID
+ * @param proposerPrivateKey The private key of the proposer
+ */
+export interface SafeTransactionProposerConfig {
+  safeAddress: Hex;
+  rpcUrl: string;
+  apiKey: string;
+  network: Network;
+  proposerPrivateKey: Hex;
+  tokenAddress: Hex;
+  contractAddress: Hex;
+}
+
+/**
+ * A class for proposing transactions to the Safe
+ * @param config The configuration for the Safe transaction proposer
+ */
+export class SafeTransactionProposer implements RewardsAllocator {
```

Comment:
This implementation was based on:
- https://docs.safe.global/core-api/transaction-service-guides/transactions
- https://safe-transaction-base-sepolia.safe.global/#/transactions/safes_multisig_transactions_create_2